buildsystem/spack/deception/spack.yaml

```diff
@@ -19,7 +19,7 @@ spack:
     exago:
       require: +raja+hiop+ipopt
     hiop:
-      require: '@1.0.0+sparse+mpi+ginkgo+kron'
+      require: '@develop+sparse+mpi+ginkgo+kron'
```

Comment:
@jaelynlitz did we get this working s.t. every rebuild actually gets a new hiop@develop? I cannot remember... 

Since you are also debugging potential hiop issues, you should make sure to maybe have a parallel hiop PR if you need to make any additional changes.

Reply:
The pipeline cloned the correct version of spack and built hiop@develop so yes?

Reply:
Right but as we found, if you build the @develop spec, spack will cache that in future builds, and so a proper rebuild might not happen unless you remove the old version, or `spack develop HiOp` with new source code

Reply:
This is a feature you worked on before iirc so I am checking